hrabsence/CRM/HRAbsence/Form/AbsenceType.php

```diff
@@ -106,11 +106,32 @@ public function buildQuickForm() {
     $this->addFormRule(array('CRM_HRAbsence_Form_AbsenceType', 'formRule'), $this);
   }
 
-  static function formRule($fields, $files, $self) {
-    $errors = array();
+  /**
+   * Used in validation from addFormRule
+   *
+   * @param array $fields
+   *  An array of fields from the form
+   * @param array $files
+   *  An array of uploaded files
+   * @param self $self
+   *  An instance of this class
+   * @return array
+   *   An array of errors
+   */
+  public static function formRule($fields, $files, $self) {
+    $errors = [];
     if (!array_key_exists('allow_debits', $fields) && !array_key_exists('allow_credits', $fields)) {
       $errors['allow_debits'] = $errors['allow_credits'] = ts("Please choose either 'Allow Debits' and/or 'Allow Credits'");
     }
+
+    $title = CRM_Utils_Array::value('title', $fields);
+    if ($self->getAction() === CRM_Core_Action::ADD) {
+      $existing = civicrm_api3('HRAbsenceType', 'get', ['name' => $title]);
```

Comment:
I can leave this for now, but it would definitely be nice to have some way to combine the logic. I think the forms offer some handy functionality - like returning multiple errors linked to each field. 

As a pure API user (like someone integrating with the site) it would be good to have an array of descriptive errors in the result, similar to what the form validation is producing.